observation/sources/Joystick.cpp

```diff
@@ -14,6 +14,7 @@
  * limitations under the License.
  */
 
+#ifndef __APPLE__
```

Comment:
We should do this selection in the build system rather than in the source file. To do it, you can go to the corresponding BUILD file (here `observation/sources/BUILD`) and do something like:

```

cc_library(
    name = "sources",
    deps = [
        ":cpu_temperature",
        select({
            "//:linux": [":joystick"],
            "@//conditions:default": [],
        }),
    ],
)
```

Reply:
Gotcha, will have a look.

Reply:
I think my last commit fixes both, let me know if it's what you meant